src/ProcessManager.php

```diff
@@ -452,29 +311,36 @@ public function run()
         $this->controller = new UnixServer($this->getControllerSocketPath(), $this->loop);
         $this->controller->on('connection', [$this, 'onSlaveConnection']);
 
-        $this->web = new Server(sprintf('%s:%d', $this->host, $this->port), $this->loop);
-        $this->web->on('connection', [$this, 'onRequest']);
+        $this->startListening();
 
         $pcntl = new \MKraemer\ReactPCNTL\PCNTL($this->loop);
         $pcntl->on(SIGTERM, [$this, 'shutdown']);
         $pcntl->on(SIGINT, [$this, 'shutdown']);
         $pcntl->on(SIGCHLD, [$this, 'handleSigchld']);
         $pcntl->on(SIGUSR1, [$this, 'restartSlaves']);
-        $pcntl->on(SIGUSR2, [$this, 'reloadSlaves']);
-
-        if ($this->isDebug()) {
-            $this->loop->addPeriodicTimer(0.5, function () {
-                $this->checkChangedFiles();
-            });
-        }
+        $pcntl->on(SIGUSR2, [$this, 'reload']);
 
         $loopClass = (new \ReflectionClass($this->loop))->getShortName();
 
-        $this->output->writeln("<info>Starting PHP-PM with {$this->slaveCount} workers, using {$loopClass} ...</info>");
+        $this->addOnceListener(self::EVENT_ALL_WORKERS_READY, function () {
+            $this->output->writeln(
+                sprintf(
+                    "<info>%d workers (starting at %d) up and ready. Application is ready at http://%s:%s/</info>",
+                    $this->config->getSlaveCount(),
+                    self::CONTROLLER_PORT+1,
+                    $this->config->getHost(),
+                    $this->config->getPort()
+                )
+            );
+        });
+
+        $this->output->writeln("<info>Starting PHP-PM with {$this->config->getSlaveCount()} workers, using {$loopClass} ...</info>");
         $this->writePid();
 
         $this->createSlaves();
 
+        $this->updateCodeReloadTimer();
```

Comment:
Why does the timer need to get "updated" here? Isn't this the initial creation?

Reply:
The hot code reload timer may need to be enabled/disabled if the debug setting is changed at runtime. This function manages that. Perhaps it needs a better method name?

Reply:
Is run() called when the config changes?

Reply:
`run()` is not called when the config changes. I don't think that is desirable since that function is bootstrapping the entire process manager as a server, the event loop, signal events, etc.